### PR TITLE
Task3 get api topics

### PR DESCRIPTION
**Pull request by Mielie**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
 const express = require("express");
 const { getTopics } = require("./controllers/topicsController");
+const { customErrorHandler } = require("./controllers/errorHandlerControllers");
 
 const app = express();
 
 app.use(express.json());
```
Comment (JVal-NC) [inline app.js]: This is not needed for this ticket and should therefore not be included, nice foresight but please remove.

File: db/seeds/seed.js [hunk 2]
```diff
@@ -39,8 +39,8 @@ const seed = ({ topicData, userData, articleData, commentData }) => {
       CREATE TABLE articles (
         article_id SERIAL PRIMARY KEY,
         title VARCHAR NOT NULL,
-        topic VARCHAR NOT NULL REFERENCES topics(slug),
-        author VARCHAR NOT NULL REFERENCES users(username),
+        topic VARCHAR NOT NULL REFERENCES topics(slug) ON DELETE CASCADE,
```
Comment (JVal-NC) [inline db/seeds/seed.js]: Nice anticipation again here but unnecessary for this ticket. Would suggest it's a change when a delete endpoint ticket is being worked on.

File: controllers/topicsController.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const { fetchTopics } = require("../models/topicsModels.js");
+
+exports.getTopics = (request, response) => {
+	return fetchTopics().then((topics) => {
+		response.status(200).send({ topics });
```
Comment (JVal-NC) [inline controllers/topicsController.js]: Catch block required for any unexpected errors.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const request = require("supertest");
+const testData = require("../db/data/test-data/");
+const seed = require("../db/seeds/seed");
+const db = require("../db/connection");
+
+beforeAll(() => seed(testData));
+
+afterAll(() => db.end());
+
+describe("/api/topics", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:11]: Good test!

File: controllers/errorHandlerControllers.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+exports.customErrorHandler = (error, request, response, next) => {
```
Comment (JVal-NC) [inline controllers/errorHandlerControllers.js:1]: Error handler for 500 errors passed on by the next statement required

**Review by JVal-NC — COMMENTED** [review]
Few comments for changes required please submit a new PR but strong start

**Review by pippaamy — COMMENTED** [review]
Hiya, thanks for that change! I would say I don't think the delete end point and functionality is relevant right now! Can we delete that and then I'm happy for you to merge after that! :)

**Pull request merged**